tests/backup/backup_test.go

```diff
@@ -4600,6 +4600,193 @@ var _ = Describe("{RestoreEncryptedAndNonEncryptedBackups}", func() {
+	var BackupLocationNameUser2 string
+	var backupLocationUID string
+	var cloudCredUID string
+	var cloudCred1UID string
+	var CloudCredUidList []string
+	var appContexts []*scheduler.Context
+	var bkpNamespaces []string
+	var clusterUid string
+	var clusterStatus api.ClusterInfo_StatusInfo_Status
+	var credName string
+	var cred1Name string
+	bkpNamespaces = make([]string, 0)
+	providers := getProviders()
+	buckets := getBucketName()
+	JustBeforeEach(func() {
+		StartTorpedoTest("SwapShareBackup",
+			"Share backup with same name between two users", nil, 0)
+		log.InfoD("Deploy applications")
+		contexts = make([]*scheduler.Context, 0)
+		for i := 0; i < Inst().GlobalScaleFactor; i++ {
+			taskName := fmt.Sprintf("%s-%d", taskNamePrefix, i)
+			appContexts = ScheduleApplications(taskName)
+			contexts = append(contexts, appContexts...)
+			for _, ctx := range appContexts {
+				ctx.ReadinessTimeout = appReadinessTimeout
+				namespace := GetAppNamespace(ctx, taskName)
+				bkpNamespaces = append(bkpNamespaces, namespace)
+			}
+		}
+	})
+	It("Share the backup with same name", func() {
+		Step("Create Users", func() {
+			log.InfoD("Creating %d users", numberOfUsers)
+			var wg sync.WaitGroup
+			for i := 1; i <= numberOfUsers; i++ {
+				userName := fmt.Sprintf("testuser%v", i)
+				firstName := fmt.Sprintf("FirstName%v", i)
+				lastName := fmt.Sprintf("LastName%v", i)
+				email := fmt.Sprintf("testuser%v@cnbu.com", i)
+				wg.Add(1)
+				go func(userName, firstName, lastName, email string) {
+					err := backup.AddUser(userName, firstName, lastName, email, "Password1")
+					log.FailOnError(err, "Failed to create user - %s", userName)
+					users = append(users, userName)
+					wg.Done()
+				}(userName, firstName, lastName, email)
+			}
+			wg.Wait()
+		})
+		Step("Adding Credentials and Registering Backup Location for testuser1 and testuser2", func() {
+			log.InfoD("Creating bucket, cloud credentials and backup location for testuser1 and testuser2")
+			for _, provider := range providers {
+				BucketName = fmt.Sprintf("%s-%v", provider, buckets[0])
+				cloudCredUID = uuid.New()
+				CloudCredUidList = append(CloudCredUidList, cloudCredUID)
+				backupLocationUID = uuid.New()
+				credName = fmt.Sprintf("autogenerated-cred-%v", time.Now().Unix())
+				backup.AddRoleToUser(users[0], backup.InfrastructureOwner, "adding admin role for testuser1")
+				backup.AddRoleToUser(users[1], backup.InfrastructureOwner, "adding admin role for testuser2")
+				UserContextBackupLocation = users[0]
+				CreateCloudCredential(provider, credName, cloudCredUID, orgID)
+				time.Sleep(time.Minute * 1)
+				BackupLocationNameUser1 = fmt.Sprintf("autogenerated-backup-location-%v", time.Now().Unix())
+				CreateBackupLocation(provider, BackupLocationNameUser1, backupLocationUID, credName, cloudCredUID, BucketName, orgID, "")
+				log.InfoD("Created Backup Location with name - %s", BackupLocationNameUser1)
+				UserContextBackupLocation = users[1]
```

Comment:
Do not use variable names starting with upper case unless you want to export it or share it across packages. Please change this everywhere.